### DISCUSSION TITLE
Are security vulnerabilities available via API?

### DISCUSSION THREAD

**Discussion by kyptin** [community/community#24516] [Apps, API and Webhooks]
<p>I realize that security vulnerability alerts can be sent via email notifications and web notifications. But I’d like to be notified in Slack instead.</p>
<p>(A natural way to do this would be via a webhook, but so far there is <a href="https://developer.github.com/webhooks/" rel="nofollow noopener">no mention of a webhook for security alerts</a>.)</p>
<p>To this end, if I have chosen to <a href="https://help.github.com/articles/choosing-the-delivery-method-for-your-notifications/" rel="nofollow noopener">receive security alerts as web notifications</a>, can I access them via the <a href="https://developer.github.com/v3/activity/notifications/" rel="nofollow noopener">Notifications API</a>? I realize I’d have to filter them out of the stream of all notifications, but my question is whether they will be present in that stream at all.</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/kyptin">@kyptin</a>,</p>
<p>Thanks for posting about this! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. Unfortunately, this isn’t currently possible, but I’ve logged your feedback in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Comment by kyptin** [comment]
<p>Thanks, <a class="mention" href="/nadiajoyce">@nadiajoyce</a>. If I’m making a feature request, it would be for a webhook, as I think that would be the most convenient. But it’s useful to know that this isn’t yet possible. Thanks again!</p>

**Comment by afeld** [comment]
<p>New API for them: <a href="https://developer.github.com/changes/2018-04-24-preview-dependency-graph-and-vulnerability-hooks/" rel="nofollow noopener">https://developer.github.com/changes/2018-04-24-preview-dependency-graph-and-vulnerability-hooks/</a></p>

**Comment by johnkors** [comment]
<p>I wondered about the same thing. Not sure if this was added at a point later on time, but in the Notifications API, there’s a _reason_ named <code>security_alert</code>.</p>
<p><a href="https://developer.github.com/v3/activity/notifications/#notification-reasons" rel="nofollow noopener">https://developer.github.com/v3/activity/notifications/#notification-reasons</a></p>

**Comment by jsegitz** [comment]
<p>Having this API is great. Would it be possible to have a scope for this API? I want to add this to our CI system, but I want the token to be only used for accessing the security alerts</p>

**Comment by phillmv** [comment]
<p>Security alert web hooks should be accessible via the repository_vulnerability_alert payload: <a href="https://developer.github.com/webhooks/event-payloads/#repository_vulnerability_alert" rel="nofollow noopener">https://developer.github.com/webhooks/event-payloads/#repository_vulnerability_alert</a></p>
